backend/pick-git/src/test/java/com/woowacourse/pickgit/authentication/application/OAuthServiceMockTest.java

```diff
@@ -0,0 +1,176 @@
+            Optional.empty());
+        when(jwtTokenProvider.createToken(githubProfileResponse.getName())).thenReturn(jwtToken);
+
+        // when
+        String token = oAuthService.createToken(code);
+
+        // then
+        assertThat(token).isEqualTo(jwtToken);
+        verify(userRepository, times(1)).findByBasicProfile_Name(githubProfileResponse.getName());
+        verify(userRepository, never()).save(user);
+        verify(jwtTokenProvider, times(1)).createToken(githubProfileResponse.getName());
+        verify(oAuthAccessTokenDao, times(1)).insert(jwtToken, oauthAccessToken);
+    }
+
+    @DisplayName("로그인(첫 로그인이 아닌경우)시 Github Profile을 가져와서 DB에 저장된 기존 정보를 update한다.")
+    @Test
+    void createToken_Signup_UpdateUserProfile() {
+        // given
+        String code = "oauth authorization code";
+        String oauthAccessToken = "oauth access token";
+        String jwtToken = "jwt token";
+
+        OAuthProfileResponse githubProfileResponse = new OAuthProfileResponse();
+        githubProfileResponse.setName("test");
+        githubProfileResponse.setDescription("hi~");
+
+        User user = new User(
+            githubProfileResponse.toBasicProfile(),
+            githubProfileResponse.toGithubProfile()
+        );
+
+        // mock
+        when(oAuthClient.getAccessToken(code)).thenReturn(oauthAccessToken);
+        when(oAuthClient.getGithubProfile(oauthAccessToken)).thenReturn(githubProfileResponse);
+        when(userRepository.findByBasicProfile_Name(githubProfileResponse.getName())).thenReturn(
+            Optional.of(user));
+        when(jwtTokenProvider.createToken(githubProfileResponse.getName())).thenReturn(jwtToken);
+
+        // when
+        String token = oAuthService.createToken(code);
+
+        // then
+        assertThat(token).isEqualTo(jwtToken);
+        verify(userRepository, times(1)).findByBasicProfile_Name(githubProfileResponse.getName());
+        verify(userRepository, times(1)).save(user);
+        verify(jwtTokenProvider, times(1)).createToken(githubProfileResponse.getName());
+        verify(oAuthAccessTokenDao, times(1)).insert(jwtToken, oauthAccessToken);
+    }
+
+    @DisplayName("JWT 토큰을 통해 AccessTokenDB에서 LoginUser에 대한 정보를 가져온다.")
+    @Test
+    void findRequestUserByToken_ValidToken_ReturnAppUser() {
+        // given
+        String token = "jwt token";
+        String accessToken = "oauth access token";
+        String username = "pick-git";
+
+        // mock
+        when(jwtTokenProvider.getPayloadByKey(token, "username")).thenReturn(username);
+        when(oAuthAccessTokenDao.findByKeyToken(token)).thenReturn(Optional.ofNullable(accessToken));
+
+        // when
+        AppUser appUser = oAuthService.findRequestUserByToken(token);
+
+        // then
+        assertThat(appUser).isInstanceOf(LoginUser.class);
+        assertThat(appUser.getUsername()).isEqualTo(username);
+        assertThat(appUser.getAccessToken()).isEqualTo(accessToken);
```

Comment:
```suggestion
        assertThat(appUser)
            .usingRecursiveComparison()
            .isEqualTo(XXX)
```

이런 방식으로 코드 변경할 수 있을 거 같아요 !